### PR TITLE
Roman’s Changes to MaxwellUtils.

### PR DESCRIPTION
**Pull request by dwfmarchant**
Roman's latest updates to MaxwellUtils.

### CODE REVIEW

**Comment by dwfmarchant** [comment]
Tests don't run - let me try and clean these up before we merge.

**Comment by Pbellive** [comment]
Sounds good Dave. Once you get the tests running could you please squash your commits so we just end up merging one commit. Thanks, Patrick.

**Comment by dwfmarchant** [comment]
I'll have to figure out haw to do that but sure.

It looks like the problem I'm running into is a circular import.   For some reason JOcTree depends on MaxwellUtils, in particular, the functions in diagM.jl.  These are not used anywhere else in MaxwellUtils.  Do you know if they are used by anything else?  I think they should be moved to a new home - maybe into JOcTree?

**Comment by Pbellive** [comment]
Re squashing: checkout this post for a quick description of how to do it: https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

Re circular dependencies: The latest version of JOcTree I have doesn't depend on MaxwellUtils. Was that dependency created by Roman? I just did a search through all the juliaInv code and I can't find anywhere that the functions in `diagM.jl` are used. There is a commented out chunk of code in `regularizeOcTree2.jl` that calls `DiagTimesM` from `diagM.jl` but that's the only place it comes up. 

In short, it doesn't make any sense to have JOcTree depend on MaxwellUtils. If Roman created the dependency to use the code in diagM.jl, then diagM.jl can just be moved to JOcTree cause it isn't called from anywhere else.

**Comment by dwfmarchant** [comment]
Yeah, it was added in his latest set of changes...  Didn't realize I was on that branch.  I'll move the particular function he need to the file where he needs it in joctree when I merge his updates there for now.

As for the tests, I don't think they ever ran in the current form.  I think we should go ahead with this merge for now and rewrite the tests some time soon.

I'll figure out how to squash these.

**Comment by Pbellive** [comment]
Sounds good to me. Those tests should get fixed soon but I think you're right, that's a separate issue that can be tackled soon. Looks like things have been squashed now. Is this ready to be merged, as far as you're concerned?

**Comment by dwfmarchant** [comment]
Yeah, I think so.
